source/shared/cpp/ObjectModel/MarkDownBlockParser.cpp

```diff
@@ -464,22 +492,12 @@ bool LinkParser::MatchAtLinkDestinationStart(std::stringstream& lookahead)
 
     if (lookahead.peek() == ')')
     {
-        lookahead.get();
-        return true;
-    }
-
-    // parses destination
-    ParseBlock(lookahead);
-
-    if (lookahead.peek() == ')')
-    {
+        ParseBlock(lookahead);
```

Comment:
this looks like a bug here.
with string `[abc]()`
if we call `ParseBlock`, it will capture ')' and the subsequent call to `MatchAtLinkDestinationRun` will fail since ')' is consumed. which isn't correct.